### ISSUE TITLE
New cop for `..` operator precedence

### ISSUE THREAD

**Issue by pocke** [rubocop/rubocop#4182]
The following code has a bug.

```ruby
# test.rb
def foo(x = nil)
  x || 1..2
end

p foo
p foo(1..3)
```

```sh
$ ruby test.rb
1..2
test.rb:3:in `foo': bad value for range (ArgumentError)
	from test.rb:7:in `<main>'
```

The code is interpreted as `(x || 1)..2`, but it looks like `x || (1..2)` to me.
I think it is confusing, so, we need a cop for this problem. However we don't have the cop.

The cause is `..` operator's precedence. https://ruby-doc.org/core-2.2.0/doc/syntax/precedence_rdoc.html

--------

## Expected behavior

```ruby
x || 1..2
```

RuboCop adds an offence for the above code.

## Actual behavior

The cop for this problem does not exist.

## RuboCop version

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
It can also be unexpected on the "other side" of the range:

```ruby
1..2.to_a
```

will attempt to send `#to_a` to `2`, rather than the range.

**Comment by bbatsov** [comment]
True. I've made this mistake like a million times. :-)

**Comment by denys281** [comment]
@Drenmi is it actual issue?
